geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -110,6 +111,42 @@ private static MemberSet calculateDiff(RegionProvider regionProvider, List<Redis
     return diff;
   }
 
+  public static Set<byte[]> sinter(RegionProvider regionProvider, List<RedisKey> keys) {
+    List<RedisSet> sets = new ArrayList<>(keys.size());
+    RedisSet smallestSet = NULL_REDIS_SET;
+
+    for (RedisKey key : keys) {
+      RedisSet redisSet = regionProvider.getTypedRedisData(REDIS_SET, key, true);
+      if (redisSet == NULL_REDIS_SET) {
+        return Collections.emptySet();
+      }
+      if (smallestSet == NULL_REDIS_SET) {
+        smallestSet = redisSet;
```

Comment:
This still ends up adding the initial "smallestSet" to "sets". You could fix this by adding an "else" to this "if".